rep-2014.rst

```diff
@@ -0,0 +1,238 @@
+
+- `Grey-box performance testing`: More application-specific measure which is capable of watching internal states of the system and can measure (probe) certain points in the system, thus generate the measurement data minimal interference. Requires to instrument the complete application.
+
+Graphically depicted:
+
+::
+
+             Probe      Probe
+             +            +
+             |            |
+    +--------|------------|-------+     +-----------------------------+
+    |        |            |       |     |                             |
+    |     +--|------------|-+     |     |                             |
+    |     |  v            v |     |     |        - latency   <--------------+ Probe
+    |     |                 |     |     |        - throughput<--------------+ Probe
+    |     |     Function    |     |     |        - memory    <--------------+ Probe
+    |     |                 |     |     |        - power     <--------------+ Probe
+    |     +-----------------+     |     |                             |
+    |      System under test      |     |       System under test     |
+    +-----------------------------+     +-----------------------------+
+
+
+              Functional                            Non-functional
+
+
+    +-------------+                     +----------------------------+
+    | Test App.   |                     |  +-----------------------+ |
+    |  + +  +  +  |                     |  |    Application        | |
+    +--|-|--|--|--+---------------+     |  |                   <------------+ Probe
+    |  | |  |  |                  |     |  +-----------------------+ |
+    |  v v  v  v                  |     |                            |
+    |     Probes                  |     |                      <------------+ Probe
+    |                             |     |                            |
+    |       System under test     |     |   System under test        |
+    |                             |     |                      <------------+ Probe
+    |                             |     |                            |
+    |                             |     |                            |
+    +-----------------------------+     +----------------------------+
+
+
+             Black-Box                            Grey-box
+
+
+Tracing and benchmarking
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Tracing and benchmarking can be defined as follows:
+
+- `tracing`: a technique used to understand what goes on in a running software system.
+
+- `benchmarking`: a method of comparing the performance of various systems by running a common test.
+
+From these definitions, inherently one can determine that both benchmarking and tracing are connected in the sense that the test/benchmark will use a series of measurements for comparison. These measurements will come from tracing probes. In other words, tracing will collect data that will then be fed into a benchmark program for comparison.
+
+
+
+
+Prior work
```

Comment:
This seems like it could be in an appendix.

Reply:
Happy to consider moving it to the appendix if you can make a contribution around it @sloretz but as is, it's going to be a significant amount of work to re-structure the document to remain coherent while moving this to the appendix.

Also, I personally think/feel this topic is important enough to give it priority in here (lots of people re-inventing the wheel benchmarking-wise. Bringing past work to their attention early in this document is sound IMHO).